KWZP2019/KWZP2019/OSHControl.cs

```diff
@@ -14,13 +14,80 @@ public partial class OSHControl : Form
     {
         private RoofingCompanyEntities db;
         private StartForm startForm;
-        private QualityControl qualityControl;
-        public OSHControl(RoofingCompanyEntities db, StartForm startForm, QualityControl qualityControl)
+        private QualityControl qualityControlForm;
+        private int selectedEmployee;
+
+        public OSHControl(RoofingCompanyEntities db, StartForm startForm, QualityControl qualityControlForm)
         {
             InitializeComponent();
             this.db = db;
             this.startForm = startForm;
-            this.qualityControl = qualityControl;
+            this.qualityControlForm = qualityControlForm;
+        }
+
+        private void btnReturn_Click(object sender, EventArgs e)
+        {
+            this.qualityControlForm.Show();
+            this.Hide();
+        }
+
+        private void btnReturnMain_Click(object sender, EventArgs e)
+        {
+            this.startForm.Show();
+            this.qualityControlForm.Close();
+            this.Close();
+        }
+
+        private void OSHControl_Load(object sender, EventArgs e)
+        {
+            dateTimeOSHControl.Value = DateTime.Now;
+            List<Employee> employeeList = db.Employees.ToList();
+            foreach (Employee employee in employeeList)
+            {
+                domUDEmployee.Items.Add(employee.IdEmployee);
+            }
+        }
+
+        private void OSHControl_FormClosed(object sender, FormClosedEventArgs e)
+        {
+            this.startForm.Show();
+            this.qualityControlForm.Close();
+        }
+
+        private void domUDEmployee_SelectedItemChanged(object sender, EventArgs e)
+        {
+            Employee employee = db.Employees.FirstOrDefault(empee => empee.IdEmployee.ToString() == domUDEmployee.Text);
+            lblEmployeeFullName.Text = employee != null ? $"{employee.EmployeeName} {employee.EmployeeSurname}" : "";
```

Comment:
czy jeśli employee będzie nullem to jest poprawne zachowanie? może powinniśmy wyświetlić jakiś błąd zamiast pustej labelki?

Reply:
tutaj employee raczej nigdy nie będzie nullem bo z tego co jest wyżej to mamy możliwość wyboru tylko z listy osób w bazie, a nie wpisywania id pracownika ręcznie, czym moglibyśmy wykroczyć poza zakres tabeli w bazie , z tego też powodu wyżej nie jest potrzebne FirstOrDefault.

Reply:
skoro nigdy nie bedzie nullem to po co sprawdzenie?

Reply:
to była podpowiedź, żeby kolega mógł poprawić